src/app/core/services/asset-create-request-builder.ts

```diff
@@ -56,6 +57,35 @@ export class AssetCreateRequestBuilder {
     const geoReferenceMethod = formValue.advanced?.geoReferenceMethod;
     const dataModel = formValue.advanced?.dataModel;
 
+    const sovereignLegalName = formValue.advanced?.sovereignLegalName;
+    const geoLocation = formValue.advanced?.geoLocation;
+    const nutsLocations = formValue.advanced?.nutsLocations;
+    const dataSampleUrls = formValue.advanced?.dataSampleUrls;
+    const referenceFileUrls = formValue.advanced?.referenceFileUrls;
+    const referenceFilesDescription =
+      formValue.advanced?.referenceFilesDescription;
+    const conditionsForUse = formValue.advanced?.conditionsForUse;
+    const dataUpdateFrequency = formValue.advanced?.dataUpdateFrequency;
+    let temporalCoverageFrom =
+      formValue.advanced?.temporalCoverage?.from || undefined;
+    let temporalCoverageToInclusive =
+      formValue.advanced?.temporalCoverage?.toInclusive || undefined;
+    /*
+     * We dont care about hours, we just want the date to be displayed identically in all timezones.
+     * For requests the hour context is lost (in both directions) as the api clients just convert the date to
+     * a field like this temporalCoverageFrom: "2024-01-01".
+     * The date string is then parsed into a date object again assuming UTC
+     * Therefore to display dates identically in all timezones we convert the date picked (which contains hour
+     * information as per timezone) to a GMT date at hour 0 here.
+     * This results to dates being displayed identically in all timezones.
+     */
```

Comment:
very nice, you could extract this method into a `date-utils.ts` file and call it "trucateToLocalDate(date: Date): Date`